ownCloud/Theming/ThemeCollection.swift

```diff
@@ -369,6 +405,62 @@ class ThemeCollection : NSObject {
 	}
 }
 
+extension NSDictionary {
```

Comment:
[optional] I think it would be good for clarity to create a new class instead of extending `NSDictionary`. That class could:
- wrap the `NSDictionary` with the colors
- also contain the `NSDictionary` with the `generics`, so they don't have to passed around for every call
- help to better understand the code/structure thanks to being a dedicated class with its own name
- be better prepared for being extended in the future (I've written lots of code repurposing `NSDictionary` in the past and have had regrets down the road about that decision in almost every case of doing that)

Reply:
Thank you for this hint. It is now better readable and with less code to pass. Put everything in new class `ThemeColorsResolver`

Reply:
Nice implementation! Thanks!